src/Symfony/Component/Security/Http/Event/LoginSuccessEvent.php

```diff
@@ -53,6 +55,7 @@ public function __construct(AuthenticatorInterface $authenticator, PassportInter
         $this->authenticator = $authenticator;
         $this->passport = $passport;
         $this->authenticatedToken = $authenticatedToken;
+        $this->previousToken = $previousToken ?? new NullToken();
```

Comment:
I would like to not use `NullToken` anywhere outside the authorization checker, as they are a major exception to the "tokens are always authenticated" rule in the new security system.

Let's make `getPreviousToken(): ?TokenInterface` instead.